File: client/registration/src/app/shared/domain/UserJob.ts
import {Job} from "./Job";
import {User} from "./User";

export class UserJob{
    id: number;
    job: Job;
    user: User;
    state: number;
    constructor(arg?:{
        id: number,
        job: Job,
        user: User,
        state: number,
    }){
        this.id = arg.id;
        this.job = arg.job;
        this.user = arg.user;
        this.state = arg.state;
    }
}
// private Long id;
// @ManyToOne
// private Job job;
// @ManyToOne
// User user;
// private Integer state; // 0 未开始 1进行中 2已结束
//
// private Double score;

File: server/src/main/java/com/shnu/part/domain/UserJob.java
package com.shnu.part.domain;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.ManyToOne;

@Entity
public class UserJob {
    @Id
    @GeneratedValue
    private Long id;
    @ManyToOne
    private Job job;
    @ManyToOne
    User user;
    private Integer state; // 0 未开始 1进行中 2已结束
    @ManyToOne
    public Score score;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Job getJob() {
        return job;
    }

    public void setJob(Job job) {
        this.job = job;
    }

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }

    public Integer getState() {
        return state;
    }

    public void setState(Integer state) {
        this.state = state;
    }

    public Score getScore() {
        return score;
    }

    public void setScore(Score score) {
        this.score = score;
    }
}
